packages/block-editor/src/components/block-tools/use-show-block-tools.js

```diff
@@ -57,7 +59,8 @@ export function useShowBlockTools() {
 				! _showEmptyBlockSideInserter && maybeShowBreadcrumb,
 			showBlockToolbarPopover: _showBlockToolbarPopover,
 			showZoomOutToolbar:
-				editorMode === 'zoom-out' &&
+				hasSelectedBlock &&
```

Comment:
So this was probably the only key change to fix the issue. The rest was what I felt were minor code quality improvements.

Reply:
Nice find!